### PR TITLE
Hl 1024 update | Adding patient validation and test case for POST /subscription-request

### PR DESCRIPTION
**Pull request by Nexengineer**

### CODE REVIEW

File: src/main/java/in/projecteka/datanotificationsubscription/DataNotificationSubscriptionConfiguration.java [hunk 1]
```diff
@@ -66,6 +71,9 @@
 @Configuration
 public class DataNotificationSubscriptionConfiguration {
 
+    @Value("${webclient.maxInMemorySize}")
```
Comment (mddubey) [inline src/main/java/in/projecteka/datanotificationsubscription/DataNotificationSubscriptionConfiguration.java:74]: We should not need this here. This was implemented on CM to fetch the QR code.

File: src/main/java/in/projecteka/datanotificationsubscription/DataNotificationSubscriptionConfiguration.java [hunk 2]
```diff
@@ -283,4 +265,76 @@ public GlobalExceptionHandler clientErrorExceptionHandler(ErrorAttributes errorA
...
+
+    @Bean("customBuilder")
+    public WebClient.Builder webClient(
+            @Qualifier("cmHttpConnector") final ClientHttpConnector clientHttpConnector,
+            ObjectMapper objectMapper) {
+        return WebClient
+                .builder()
+                .exchangeStrategies(exchangeStrategies(objectMapper))
+                .clientConnector(clientHttpConnector)
+                .codecs(configurer -> configurer
```
Comment (mddubey) [inline src/main/java/in/projecteka/datanotificationsubscription/DataNotificationSubscriptionConfiguration.java:308]: We can remove the codecs part.

File: src/main/java/in/projecteka/datanotificationsubscription/UserServiceProperties.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.ConstructorBinding;
+
+@ConfigurationProperties(prefix = "subscriptionmanager.userservice")
+@Getter
+@AllArgsConstructor
+@ConstructorBinding
+@Builder
+public class UserServiceProperties {
+    private final String url;
```
Comment (mddubey) [inline src/main/java/in/projecteka/datanotificationsubscription/UserServiceProperties.java:15]: I think we should be able to get rid of most of the properties here. We will not use anything related to otp_attempt, health_id_search_attempt, enable_share.

File: src/main/java/in/projecteka/datanotificationsubscription/DataNotificationSubscriptionConfiguration.java (hunk 1 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/datanotificationsubscription/DataNotificationSubscriptionConfiguration.java:74]: Removed

File: src/main/java/in/projecteka/datanotificationsubscription/DataNotificationSubscriptionConfiguration.java (hunk 2 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/datanotificationsubscription/DataNotificationSubscriptionConfiguration.java:308]: removed

File: src/main/java/in/projecteka/datanotificationsubscription/clients/model/Session.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Session {
```
Comment (mddubey) [inline src/main/java/in/projecteka/datanotificationsubscription/clients/model/Session.java:15]: There will already be a Session Model. If possible we can use that itself.

File: src/main/java/in/projecteka/datanotificationsubscription/UserServiceProperties.java (hunk 3 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/datanotificationsubscription/UserServiceProperties.java:15]: Done

File: src/main/java/in/projecteka/datanotificationsubscription/clients/model/Session.java (hunk 4 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/datanotificationsubscription/clients/model/Session.java:15]: Done

**Review by mddubey — APPROVED** [review]

**Pull request merged**